apps/dashboard/src/app/(dashboard)/(chain)/[chain_id]/(chainPage)/components/client/FaucetButton.tsx

```diff
@@ -0,0 +1,146 @@
+"use client";
+
+import { Spinner } from "@/components/ui/Spinner/Spinner";
+import { Button } from "@/components/ui/button";
+import { THIRDWEB_ENGINE_FAUCET_WALLET } from "@/constants/env";
+import { CustomConnectWallet } from "@3rdweb-sdk/react/components/connect-wallet";
+import { useMutation } from "@tanstack/react-query";
+import { thirdwebClient } from "lib/thirdweb-client";
+import { useRouter } from "next/navigation";
+import { toast } from "sonner";
+import { defineChain, toUnits } from "thirdweb";
+import type { ChainMetadata } from "thirdweb/chains";
+import { useActiveAccount, useWalletBalance } from "thirdweb/react";
+
+function formatTime(seconds: number) {
+  const rtf = new Intl.RelativeTimeFormat("en", { numeric: "auto" });
+  const hours = Math.floor(seconds / 3600);
+  const minutes = Math.floor((seconds % 3600) / 60);
+  if (hours > 0) {
+    return rtf.format(+hours, "hour");
+  }
+
+  if (minutes > 0) {
+    return rtf.format(+minutes, "minute");
+  }
+
+  return rtf.format(+seconds, "second");
+}
+
+export function FaucetButton({
+  chain,
+  ttlSeconds,
+  amount,
+}: {
+  chain: ChainMetadata;
+  ttlSeconds: number;
+  amount: string;
+}) {
+  const address = useActiveAccount()?.address;
+  const chainId = chain.chainId;
+  const definedChain = defineChain(chainId);
+  const faucetWalletBalanceQuery = useWalletBalance({
+    address: THIRDWEB_ENGINE_FAUCET_WALLET,
+    chain: definedChain,
+    client: thirdwebClient,
+  });
+  // for some bizzare reason, If I use useMutation here, it's never executing the mutationFn !!!
+  const claimMutation = useMutation({
+    mutationFn: async () => {
+      const response = await fetch("/api/testnet-faucet/claim", {
+        method: "POST",
+        headers: {
+          "Content-Type": "application/json",
+        },
+        body: JSON.stringify({
+          chainId: chainId,
+          toAddress: address,
+          amount,
+        }),
```

Comment:
the address should already be available based on the auth cookie on the backend since I assume this request has to be authenticated - if not why not?

Reply:
if its already an authenticated request just infer the address from the backend imo

Reply:
I haven't looked in how the Faucet is working - just changed the design. Let me check

Reply:
On this page, we are not enforcing the users to sign in - they can just connect wallet and go

Reply:
feels a bit odd in general for a faucet but that's OK for now